@here/harp-mapview/lib/ScreenCollisions.ts

```diff
@@ -104,6 +168,107 @@ export class ScreenCollisions {
     isVisible(bounds: Math2D.Box): boolean {
         return this.screenBounds.intersects(bounds);
     }
+
+    private intersectsLine(bbox: CollisionBox, boundedLine: LineWithBound): boolean {
+        const line = boundedLine.line;
+        // Transform line from screen space to screenBounds space.
+        const lineStartXTransformed = line.start.x + this.screenBounds.x;
+        const lineStartYTransformed = this.screenBounds.h / 2 - line.start.y;
+        const lineEndXTransformed = line.end.x + this.screenBounds.x;
+        const lineEndYTransformed = this.screenBounds.h / 2 - line.end.y;
+
+        // Test left side
+        let intersectionResult = checkIntersection(
```

Comment:
Intersecting every box edge with the line segment would not catch the case when the box is completely containing the line segment.
I would go for another approach, i.e. check all the box corners with the infinite line equation(not the line segment) and if not all corners are on one side you have an intersection (this only works b/c we check if the bbox of the line interscets with the box before)

Reply:
This is a great idea, I will give it a try.